src/bp-forums/core/options.php

```diff
@@ -486,6 +486,8 @@ function bbp_maybe_get_root_slug() {
 	if ( bbp_get_root_slug() && bbp_include_root_slug() )
 		$retval = trailingslashit( bbp_get_root_slug() );
 
+	$retval = urldecode_deep( $retval) ;
+	
```

Comment:
@janakkaneriya007 

```
L486: Generic.ControlStructures.InlineControlStructure.NotAllowed: Inline control structures are not allowed
L489: PEAR.Functions.FunctionCallSignature.SpaceAfterCloseBracket: Space after closing parenthesis of function call prohibited
L489: PEAR.Functions.FunctionCallSignature.SpaceBeforeCloseBracket: Expected 1 spaces before closing parenthesis; 0 found
L489: Squiz.WhiteSpace.SemicolonSpacing.Incorrect: Space found before semicolon; expected ");" but found ") ;"
L490: Squiz.WhiteSpace.SuperfluousWhitespace.EndLine: Whitespace found at end of line
```

Reply:
@janakkaneriya007 

1. what if `$retval` is empty string?
2. what if function return value from filter `bbp_maybe_get_root_slug` is not urldecoded.